front/src/applications/stdcmV2/components/StdcmInputVia.tsx

```diff
@@ -20,27 +24,68 @@ const StdcmInputVia = ({
     pathStep.stopFor ? `${ISO8601Duration2sec(pathStep.stopFor) / 60}` : ''
   );
 
+  const [showWarning, setShowWarning] = useState(false);
+
   const debounceUpdatePathStepStopTime = useMemo(
     () => debounce((value) => updatePathStepStopTime(value), 500),
     []
   );
 
+  useEffect(() => {
+    let newStopTime = pathStepStopTime;
+    let warning = false;
+
+    const isDriverSwitch = stopType === StdcmStopTypes.DRIVER_SWITCH;
+    const isPassageTime = stopType === StdcmStopTypes.PASSAGE_TIME;
+    const isServiceStop = stopType === StdcmStopTypes.SERVICE_STOP;
+
+    if (isPassageTime && pathStepStopTime !== '0') {
+      newStopTime = '0';
+    } else if (isDriverSwitch) {
+      const stopTimeValue = Number(pathStepStopTime);
+      if (stopTimeValue === 0) {
+        newStopTime = '3';
+      } else if (stopTimeValue < 3) {
+        warning = true;
+      }
+    } else if (isServiceStop && pathStepStopTime <= '0') {
+      newStopTime = '1';
+    }
+
+    if (newStopTime !== pathStepStopTime) {
+      setPathStepStopTime(newStopTime);
+      updatePathStepStopTime(newStopTime);
+    }
+
+    setShowWarning(warning);
+  }, [stopType, pathStepStopTime]);
```

Comment:
Not sure it's healthy to have in the dependency array the state and update with it's setter inside, it might go through some loops

Reply:
I had to keep it but I moved lot of the logic in parent file.